Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -1335,29 +1335,35 @@ void RenderLayer::updateTransformFromStyle(TransformationMatrix& transform, cons
     makeMatrixRenderable(transform, canRender3DTransforms());
 }
 
-void RenderLayer::setReferenceBoxForPathOperations()
+bool RenderLayer::setReferenceBoxForPathOperations()
 {
     auto pathOperation = renderer().style().offsetPath();
     if (!pathOperation)
-        return;
+        return true;
     if (is<BoxPathOperation>(pathOperation)) {
         auto& boxPathOperation = downcast<BoxPathOperation>(*pathOperation);
         auto pathReferenceBoxRect = snapRectToDevicePixelsIfNeeded(renderer().referenceBoxRect(boxPathOperation.referenceBox()), renderer());
         boxPathOperation.setPathForReferenceRect(FloatRoundedRect { pathReferenceBoxRect });
     } else if (is<RayPathOperation>(pathOperation)) {
-        if (const auto* containingBlock = renderer().containingBlock()) {
+        if (auto* containingBlock = renderer().containingBlock()) {
             auto& rayPathOperation = downcast<RayPathOperation>(*pathOperation);
             auto pathReferenceBoxRect = snapRectToDevicePixelsIfNeeded(containingBlock->transformReferenceBoxRect(containingBlock->style()), renderer());
-            if (!pathReferenceBoxRect.width())
-                pathReferenceBoxRect.setWidth(pathReferenceBoxRect.height());
-            if (!pathReferenceBoxRect.height())
-                pathReferenceBoxRect.setHeight(pathReferenceBoxRect.width());
+
+            // If the containing block has an incomplete rect this means it is mid layout. Have the parent re-run this
+            // after completing layout
+            if (!pathReferenceBoxRect.width() || !pathReferenceBoxRect.height()) {
```

Comment:
Just checking for zero width or height seems wrong. What if it was laid out once before, but the size is changing?

Reply:
Right, we should check if the parent needs layout.